src/block-data-providers/esplora/provider.ts

```diff
@@ -144,36 +149,45 @@ export class EsploraProvider
-                        const tx = await this.getTx(txid);
-                        const vin: TransactionInput[] = tx.vin.map((input) => ({
-                            txid: input.txid,
-                            vout: input.vout,
-                            scriptSig: input.scriptsig,
-                            prevOutScript: input.prevout.scriptpubkey,
-                            witness: input.witness,
-                        }));
-                        const vout = tx.vout.map((output) => ({
-                            scriptPubKey: output.scriptpubkey,
-                            value: output.value,
-                        }));
-
-                        await this.indexTransaction(
-                            txid,
-                            vin,
-                            vout,
-                            height,
-                            hash,
-                        );
-                    }, this),
-                );
+                await this.dbTransactionService.execute(async (manager) => {
+                    await Promise.all(
+                        batch.map(async (txid) => {
+                            const tx = await this.getTx(txid);
+                            const vin: TransactionInput[] = tx.vin.map(
+                                (input) => ({
+                                    txid: input.txid,
+                                    vout: input.vout,
+                                    scriptSig: input.scriptsig,
+                                    prevOutScript: input.prevout.scriptpubkey,
+                                    witness: input.witness,
+                                }),
+                            );
+                            const vout = tx.vout.map((output) => ({
+                                scriptPubKey: output.scriptpubkey,
+                                value: output.value,
+                            }));
+
+                            await this.indexTransaction(
+                                txid,
+                                vin,
+                                vout,
+                                height,
+                                hash,
+                                manager,
+                            );
+                        }, this),
+                    );
 
-                state.indexedBlockHeight = height;
-                state.lastProcessedTxIndex = i + this.batchSize - 1;
-                await this.setState(state, {
-                    blockHeight: height,
-                    blockHash: hash,
+                    state.indexedBlockHeight = height;
+                    state.lastProcessedTxIndex = i + this.batchSize - 1;
+                    await this.setState(
+                        state,
+                        {
+                            blockHeight: height,
+                            blockHash: hash,
+                        },
+                        manager,
+                    );
```

Comment:
We are doing computation, even making a network call inside the transaction. This is not ideal and needs to be updated either in this PR or a followup PR.

So all the computation should happen outside of the transaction, and in the transaction we just update the entities atomically

Reply:
okay, we should do it in a followup PR because the method signatures will change again